### ISSUE TITLE
shadow file names

### ISSUE THREAD

**Issue by convas** [Fish-Git/hyperion#9]
Hi, I recently upgraded to 4.0.0.**8745** and the shadow file names seem to have changed.

I use:

    6C15  3390  D:/Hercules/MVS/ZOS/DLB001.3390-9.comp-z nofulltrackio sf=D:/Hercules/MVS/ZOS/DLB001_Shadow_0.3390-9.comp-z cu=3990-3

and in prior versions (4.00.0.**8143**) the shadow file were named `DLB001_Shadow_1.3390-9.comp-z`

With the current version of Hyperion I am now getting shadow files named `DLB001_Shadow_0.3390-1.comp-z`.

The shadow number is going after the `-` (dash) instead of the `_` (underscore).

**Comment by convas** [comment]
Update I was using 4.00.0.8143 before.

**Comment by Fish-Git** [comment]
@convas:

The current shadow file naming behavior you describe, while understandably unexpected, is nonetheless correct.

Shadow file naming was corrected back on _**July 19, 2015**_ by commit eb05741ac9364288f340f0363c9d6931dce8882d to match the published documentation, causing shadow file names to unexpectedly change for users using shadow file names with more than one period in them. Users whose shadow file names did not contain any extra periods were not affected.

Unfortunately you happen to be one of the users that was affected. For that I apologize.

The rule regarding the placement of the shadow file _number_ in a given shadow file name has _**always**_ been to place it immediately prior to the **_last_** period of the name (i.e. immediately before the period preceding the filename extension (or at the last character of the filename if there is no extension).

Earlier versions of Hercules (including the 3.x spinhawk series) contained a day one bug that was erroneously placing the shadow file number immediately before the _first_ period in the filename.

This was fixed in Hyperion back on July 19, 2015 by commit eb05741ac9364288f340f0363c9d6931dce8882d to match the published documentation, which was the behavior users were told to expect.

The version you were previously using (4.00.0.**8143**) did not yet contain this fix. The current version of Hyperion however _does_ contain this fix, which is why you are now experiencing the behavior you are experiencing. The behavior you are now experiencing is actually the _correct_ behavior, whereas the behavior you _were_ experiencing with your older version of Hyperion was actually _incorrect_ behavior that did not match the published documentation.

This was _not_ a documentation error. It was an _implementation_ error. The documentation was actually correct. The code that tried to implement it however was not correct (but was fixed as I said back on July 19, 2015).

I do apologize for whatever inconvenience this issue may have caused you, but I'm afraid there is nothing to be fixed. The current version of Hyperion is actually working as designed.

Nevertheless I have now made some additional changes to the documentation to provide a more detailed explanation of how shadow file naming actually works along with some working examples:

   *  [Shadow File naming](https://fish-git.github.io/html/cckddasd.html#shadowfiles)

My hope is that it might help prevent others from tripping over this unexpected change in behavior.

Thank you for reporting it.
